essential_reading/DataSources/Data_S3.md

```diff
@@ -0,0 +1,223 @@
+
+## Anonymous Reads
+
+A lot of data is available for global read, which does not require credentials or a profile.
+In this case, we set `anon=True` when plumbing the `fsspec` object:
+
+```python
+fs = fsspec.filesystem(
+    's3',
+    anon=True   # Does not require credentials
+    )
+# 'fs' is an object which provides methods access to the virtual filesystem
+# 'ls' method == list ; list files in a virtual folder to test if we really have access
+fs.ls('s3://noaa-nwm-retrospective-2-1-zarr-pds/')
+```
+
+Output:
+
+```text
+['noaa-nwm-retrospective-2-1-zarr-pds/chrtout.zarr',
+ 'noaa-nwm-retrospective-2-1-zarr-pds/gwout.zarr',
+ 'noaa-nwm-retrospective-2-1-zarr-pds/lakeout.zarr',
+ 'noaa-nwm-retrospective-2-1-zarr-pds/ldasout.zarr',
+ 'noaa-nwm-retrospective-2-1-zarr-pds/precip.zarr',
+ 'noaa-nwm-retrospective-2-1-zarr-pds/rtout.zarr']
+```
+
+We can call other methods on the `fsspec` object "`fs`" to interact with filesystem objects.
+
+```python
+fs.info('s3://noaa-nwm-retrospective-2-1-zarr-pds/index.html')
+```
+
+Output:
+
+```text
+{'Key': 'noaa-nwm-retrospective-2-1-zarr-pds/index.html',
+'LastModified': datetime.datetime(2021, 10, 1, 20, 48, 48, tzinfo=tzutc()),
+'ETag': '"3b4b4277037c1127ed4ba68e26461b2e"',
+'Size': 32357,
+'StorageClass': 'STANDARD',
+'type': 'file',
+'size': 32357,
+'name': 'noaa-nwm-retrospective-2-1-zarr-pds/index.html'}
+```
+
+Low-level operations with `fsspec` are designed to behave similarly to the 'native' file operations in Python.
+
+```python
+# Use open() to get something that behaves like a file handle for
+# low-level Python read/write operations:
+with fs.open('s3://noaa-nwm-retrospective-2-1-zarr-pds/index.html') as f:
+    # print first 5 lines...
+    for i in range(0, 5):
+        line = f.readline()
+        print(line)
+```
+
+Output:
+
+```text
+b'<!DOCTYPE html>\r\n'
+b'\r\n'
+b'<!--\r\n'
+b'Copyright 2014-2018 Amazon.com, Inc. or its affiliates. All Rights Reserved.\r\n'
+b'\r\n'
+```
+
+However -- most of the time, we don't need such low level access.
+A more general, and useful, mechanism is to '_map_' an S3 object to a _virtual file_ using `get_mapper()`.
+That _map_ object can then be treated as if it is a local file -- the `fsspec` mapping mechanism handles
+all of the translation to interact with S3.
+
+```python
+m = fs.get_mapper('s3://noaa-nwm-retrospective-2-1-zarr-pds/chrtout.zarr')
+# 'm' looks like a file as far as Python is concerned.
+# It is now a stand-in for those functions that require a file as argument.
+g = zarr.convenience.open_consolidated(m) # read zarr metadata for named file.
+print(g.tree())
+```
+
+Output:
+
+```text
+/
+ ├── crs () |S1
+ ├── elevation (2776738,) float32
+ ├── feature_id (2776738,) int32
+ ├── gage_id (2776738,) |S15
+ ├── latitude (2776738,) float32
+ ├── longitude (2776738,) float32
+ ├── order (2776738,) int32
+ ├── streamflow (367439, 2776738) int32
+ ├── time (367439,) int64
+ └── velocity (367439, 2776738) int32
+```
+
+:::{warning}
```

Comment:
when I view this as rendered markdown, these are still just showing up as text

Reply:
When you say "these" -- I think you are referring to `{warning}` ... if that's not right, then I need to understand more about your observation. 

The `warning` directive is not being parsed by the markdown renderer. That's correct -- because it isn't markdown.  This is a **rST** directive to be used when the notebook is assembled by jupyterbook. 

This may help: <https://github.com/hytest-org/hytest/blob/gt-118-JupyterBook-TOC/doc/JupyterBook_Admin.md> 

Some features in the fully-built jupterbook will not render in vanilla markdown.  The directives are constructed such that they will still render as text (just not very pretty, as you observed) in a markdown context, but will have full features when jupyterbook parses them.

Reply:
yes - sorry for the ambiguity, I was referring to :::{warning} and :::{sidebar}

It would be nice if the files rendered correctly in markdown because it seems likely that I would just send someone a link to this markdown file on our Github to read....but maybe I will be convinced otherwise once we have the JupyterBook working. I would say, leave as is for now, and once we merge in the JupyterBook TOC, we can revisit if these headers not rendering on the .md file is a concern. I am trying to get through the PRs today if possible!

Reply:
Is there an advantage to sending someone a link to this markdown on github vs sending them a link to the fully-featured HTML that jupyterbook produces from it (also on github)?

Reply:
If we are not going to send people to the fully-featured, rendered jupyterbook, then why go to the trouble of producing it?  If vanilla markdown meets the mission of the project, we can save a lot of complexity by staying within its limitations and avoid jupyterbook altogether.

Reply:
Hopefully we will always send links to the Jupyterbook, but I've never worked on a project that produced one before, so I'm just acknowledging my ignorance of any nuances that may come with it. I think we can assume we will use the JupyterBook moving forward, but I'm making a mental note to think about it a little more once we have it :)